libsolidity/analysis/ControlFlowGraph.cpp

```diff
@@ -35,10 +35,22 @@ bool CFG::constructFlow(ASTNode const& _astRoot)
 bool CFG::visit(FunctionDefinition const& _function)
 {
 	if (_function.isImplemented())
-		m_functionControlFlow[&_function] = ControlFlowBuilder::createFunctionFlow(m_nodeContainer, _function);
+		m_functionControlFlow[&_function] = ControlFlowBuilder::createFunctionFlow(m_nodeContainer, _function, m_contract);
 	return false;
 }
 
+bool CFG::visit(ContractDefinition const& _contract)
+{
+	m_contract = &_contract;
```

Comment:
I don't see how this can work as it should.
Are you visiting all other contracts for each contract in the inheritance hierarchy?

Reply:
You're right, the logic is wrong here. I'll do just that now (visiting each in the hierarchy)